Compile errors after running make style-code-android

I added `lodash` to `devDependencies` in `package.json` over in https://github.com/mapbox/mapbox-gl-native/pull/6687/commits/a9ad6539b917831af36eba7b4bbc98ef8fdac493 (in https://github.com/mapbox/mapbox-gl-native/pull/6687, the pull request I'm currently working on - hasn't been merged to `master` yet).

In #6687, running `make style-code-android` generates a diff, I think because of the logic in [`global.propertyNativeType`](https://github.com/mapbox/mapbox-gl-native/blob/242c7f8fe62eb863cf300855aee2f04022311288/platform/android/scripts/generate-style-code.js#L103-L108). It creates duplicate function signatures for a handful of `TranslateAnchorType` and `AlignmentType` properties, which cause compile errors when running `make android`.

~~**Oddly, I don't see this behavior happening on the Bitrise CI tests - does `make style-code-android` not run?**~~ I wasn't seeing this before, but now it's showing up in https://www.bitrise.io/build/ed808c6680dc1725

```
../../../platform/android/src/style/layers/../conversion/types.hpp:124:8: error: redefinition of 'mbgl::android::conversion::Converter<jni::jobject *, mbgl::style::TranslateAnchorType>'
struct Converter<jni::jobject*, mbgl::style::TranslateAnchorType> {
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../platform/android/src/style/layers/../conversion/types.hpp:96:8: note: previous definition is here
struct Converter<jni::jobject*, mbgl::style::TranslateAnchorType> {
       ^
8 errors generated.
```

```
diff --git a/platform/android/src/style/conversion/types.hpp b/platform/android/src/style/conversion/types.hpp
index d3c12ff..cc514a8 100644
--- a/platform/android/src/style/conversion/types.hpp
+++ b/platform/android/src/style/conversion/types.hpp
@@ -58,6 +58,20 @@ struct Converter<jni::jobject*, mbgl::style::IconTextFitType> {
 };
 
 template <>
+struct Converter<jni::jobject*, mbgl::style::AlignmentType> {
+    Result<jni::jobject*> operator()(jni::JNIEnv& env, const mbgl::style::AlignmentType& value) const {
+        return convert<jni::jobject*, std::string>(env, toString(value));
+    }
+};
+
+template <>
+struct Converter<jni::jobject*, mbgl::style::AlignmentType> {
+    Result<jni::jobject*> operator()(jni::JNIEnv& env, const mbgl::style::AlignmentType& value) const {
+        return convert<jni::jobject*, std::string>(env, toString(value));
+    }
+};
+
+template <>
 struct Converter<jni::jobject*, mbgl::style::TextJustifyType> {
     Result<jni::jobject*> operator()(jni::JNIEnv& env, const mbgl::style::TextJustifyType& value) const {
         return convert<jni::jobject*, std::string>(env, toString(value));
@@ -86,6 +100,34 @@ struct Converter<jni::jobject*, mbgl::style::TranslateAnchorType> {
 };
 
 template <>
+struct Converter<jni::jobject*, mbgl::style::TranslateAnchorType> {
+    Result<jni::jobject*> operator()(jni::JNIEnv& env, const mbgl::style::TranslateAnchorType& value) const {
+        return convert<jni::jobject*, std::string>(env, toString(value));
+    }
+};
+
+template <>
+struct Converter<jni::jobject*, mbgl::style::TranslateAnchorType> {
+    Result<jni::jobject*> operator()(jni::JNIEnv& env, const mbgl::style::TranslateAnchorType& value) const {
+        return convert<jni::jobject*, std::string>(env, toString(value));
+    }
+};
+
+template <>
+struct Converter<jni::jobject*, mbgl::style::TranslateAnchorType> {
+    Result<jni::jobject*> operator()(jni::JNIEnv& env, const mbgl::style::TranslateAnchorType& value) const {
+        return convert<jni::jobject*, std::string>(env, toString(value));
+    }
+};
+
+template <>
+struct Converter<jni::jobject*, mbgl::style::TranslateAnchorType> {
+    Result<jni::jobject*> operator()(jni::JNIEnv& env, const mbgl::style::TranslateAnchorType& value) const {
+        return convert<jni::jobject*, std::string>(env, toString(value));
+    }
+};
+
+template <>
 struct Converter<jni::jobject*, mbgl::style::CirclePitchScaleType> {
     Result<jni::jobject*> operator()(jni::JNIEnv& env, const mbgl::style::CirclePitchScaleType& value) const {
         return convert<jni::jobject*, std::string>(env, toString(value));
diff --git a/platform/android/src/style/conversion/types_string_values.hpp b/platform/android/src/style/conversion/types_string_values.hpp
index b42ca52..0dba7b3 100644
--- a/platform/android/src/style/conversion/types_string_values.hpp
+++ b/platform/android/src/style/conversion/types_string_values.hpp
@@ -109,6 +109,40 @@ namespace conversion {
         }
     }
 
+    //text-pitch-alignment
+    inline std::string toString(mbgl::style::AlignmentType value) {
+        switch (value) {
+          case mbgl::style::AlignmentType::Map:
+            return "map";
+            break;
+          case mbgl::style::AlignmentType::Viewport:
+            return "viewport";
+            break;
+          case mbgl::style::AlignmentType::Auto:
+            return "auto";
+            break;
+          default:
+            throw std::runtime_error("Not implemented");
+        }
+    }
+
+    //text-rotation-alignment
+    inline std::string toString(mbgl::style::AlignmentType value) {
+        switch (value) {
+          case mbgl::style::AlignmentType::Map:
+            return "map";
+            break;
+          case mbgl::style::AlignmentType::Viewport:
+            return "viewport";
+            break;
+          case mbgl::style::AlignmentType::Auto:
+            return "auto";
+            break;
+          default:
+            throw std::runtime_error("Not implemented");
+        }
+    }
+
     //text-justify
     inline std::string toString(mbgl::style::TextJustifyType value) {
         switch (value) {
@@ -192,6 +226,62 @@ namespace conversion {
         }
     }
 
+    //line-translate-anchor
+    inline std::string toString(mbgl::style::TranslateAnchorType value) {
+        switch (value) {
+          case mbgl::style::TranslateAnchorType::Map:
+            return "map";
+            break;
+          case mbgl::style::TranslateAnchorType::Viewport:
+            return "viewport";
+            break;
+          default:
+            throw std::runtime_error("Not implemented");
+        }
+    }
+
+    //icon-translate-anchor
+    inline std::string toString(mbgl::style::TranslateAnchorType value) {
+        switch (value) {
+          case mbgl::style::TranslateAnchorType::Map:
+            return "map";
+            break;
+          case mbgl::style::TranslateAnchorType::Viewport:
+            return "viewport";
+            break;
+          default:
+            throw std::runtime_error("Not implemented");
+        }
+    }
+
+    //text-translate-anchor
+    inline std::string toString(mbgl::style::TranslateAnchorType value) {
+        switch (value) {
+          case mbgl::style::TranslateAnchorType::Map:
+            return "map";
+            break;
+          case mbgl::style::TranslateAnchorType::Viewport:
+            return "viewport";
+            break;
+          default:
+            throw std::runtime_error("Not implemented");
+        }
+    }
+
+    //circle-translate-anchor
+    inline std::string toString(mbgl::style::TranslateAnchorType value) {
+        switch (value) {
+          case mbgl::style::TranslateAnchorType::Map:
+            return "map";
+            break;
+          case mbgl::style::TranslateAnchorType::Viewport:
+            return "viewport";
+            break;
+          default:
+            throw std::runtime_error("Not implemented");
+        }
+    }
+
     //circle-pitch-scale
     inline std::string toString(mbgl::style::CirclePitchScaleType value) {
         switch (value) {
```

---

**Platform:** macOS, Ubuntu 16.04 (Bitrise)
**Mapbox SDK version:** #6687

/cc @ivovandongen @mapbox/android

--- a/package.json
+++ b/package.json
@@ -20,6 +20,7 @@
     "csscolorparser": "^1.0.2",
     "ejs": "^2.4.1",
     "express": "^4.11.1",
+    "lodash": "^4.16.4",
     "mapbox-gl-shaders": "mapbox/mapbox-gl-shaders#98a56d538b11fb331aa67a6d632d6ecd6821b007",
     "mapbox-gl-style-spec": "mapbox/mapbox-gl-style-spec#7f62a4fc9f21e619824d68abbc4b03cbc1685572",
     "mapbox-gl-test-suite": "mapbox/mapbox-gl-test-suite#6df3b27868cc00432197e3fc9c166fdba7067913",

--- a/platform/android/scripts/generate-style-code.js
+++ b/platform/android/scripts/generate-style-code.js
@@ -265,7 +265,7 @@ fs.writeFileSync(
 );
 
 //De-duplicate enum properties before processing jni property templates
-const enumPropertiesDeDup = _(enumProperties).uniq(global.propertyNativeType).value();
+const enumPropertiesDeDup = _(enumProperties).uniqBy(global.propertyNativeType).value();
 
 // JNI Enum property conversion templates
 const enumPropertyHppTypeStringValueTemplate = ejs.compile(fs.readFileSync('platform/android/src/style/conversion/types_string_values.hpp.ejs', 'utf8'), {strict: true});

